File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dct = {}
        for i in range(len(nums)):
            cur_num = nums[i]
            remainder = target - cur_num
            if remainder in dct:
                return [i, dct[remainder]]
            dct[cur_num] = i
        return None

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const dict = {};

  for (let i in nums) {
    let remainder = target - nums[i];
    if (dict[remainder]) {
      return [Number(i), Number(dict[remainder])];
    } else {
      dict[nums[i]] = i;
    }
  }
  return null;
};
